src/features/auth/store/auth.slice.ts

```diff
@@ -2,31 +2,77 @@ import { createSlice } from '@reduxjs/toolkit';
 import type { RootState } from '@metis/store/types';
 import { login } from './auth.actions';
 
+type UserData = {
+  password: string;
+  passphrase: string;
+  privateKeyArmored: string;
+  publicKeyArmored: string;
+};
+
+type JupAccount = {
+  address: string;
+  alias: string;
+};
+
 export type AuthState = {
-  isLoading: boolean;
   isLoggedIn: boolean;
+  isConnectingToMetamask: boolean;
+  isCreatingAccount: boolean;
+  userData: UserData;
+  jupAccount: JupAccount;
 };
 
 const authSlice = createSlice({
   name: 'auth',
   initialState: {
-    isLoading: false,
-    isLoggedIn: true, // TODO: Make this value false when login is fully implemented
+    isLoggedIn: false,
+    isConnectingToMetamask: false,
+    isCreatingAccount: false,
+    userData: {
+      password: '',
+      passphrase: '',
+      privateKeyArmored: '',
+      publicKeyArmored: '',
+    },
+    jupAccount: {
+      address: '',
+      alias: '',
+    },
   } as AuthState,
-  reducers: {},
+  reducers: {
+    setLoggedIn: (state, { payload }) => {
+      state.isLoggedIn = payload;
+    },
+    setJupAccount: (state, { payload }) => {
+      state.jupAccount = payload;
+    },
+    setIsConnectingToMetamask: (state, { payload }) => {
+      state.isConnectingToMetamask = payload;
+    },
+    setIsCreatingAccount: (state, { payload }) => {
+      state.isCreatingAccount = payload;
+    },
+  },
   extraReducers: (builder) => {
-    builder.addCase(login.pending, (state) => {
-      state.isLoading = true;
-    });
     builder.addCase(login.fulfilled, (state, { payload }) => {
-      state.isLoading = false;
-      state.isLoggedIn = payload;
+      state.isConnectingToMetamask = false;
+      state.isCreatingAccount = true;
+      const { password, passphrase, privateKeyArmored, publicKeyArmored } = payload;
+      state.userData = {
+        ...state.userData,
```

Comment:
you could safely remove this line, or if you are getting the same shape from payload, you could easily assign the state.userData = payload